scripts/run_performance_job.py

```diff
@@ -50,258 +50,613 @@ def output_counters_for_crank(reports: list[Any]):
+    os_sub_group: Optional[str] = None
+    project_file: Optional[str] = None
+    partition_count: Optional[int] = None
     build_repository_name: str = os.environ.get("BUILD_REPOSITORY_NAME", "dotnet/performance")
     build_source_branch: str = os.environ.get("BUILD_SOURCEBRANCH", "main")
     build_number: str = os.environ.get("BUILD_BUILDNUMBER", "local")
-    internal: bool = True
-    pgo_run_type: str | None = None
-    physical_promotion_run_type: str | None = None
-    r2r_run_type: str | None = None
-    experiment_name: str | None = None
+    build_definition_name: Optional[str] = os.environ.get("BUILD_DEFINITIONNAME")
+    internal: bool = False
+    pgo_run_type: Optional[str] = None
+    physical_promotion_run_type: Optional[str] = None
+    r2r_run_type: Optional[str] = None
+    experiment_name: Optional[str] = None
     codegen_type: str = "JIT"
     runtime_type: str = "coreclr"
-    affinity: str | None = "0"
-    run_env_vars: dict[str, str] = field(default_factory=dict[str, str])
+    affinity: Optional[str] = "0"
+    run_env_vars: Dict[str, str] = field(default_factory=dict) # type: ignore
     is_scenario: bool = False
-    runtime_flavor: str | None = None
+    runtime_flavor: Optional[str] = None
     local_build: bool = False
+    compare: bool = False
+    only_sanity_check: bool = False
+    ios_llvm_build: bool = False
+    ios_strip_symbols: bool = False
+    hybrid_globalization: bool = False
+    javascript_engine: str = "NoJS"
+    send_to_helix: bool = False
+    channel: Optional[str] = None
+    perf_repo_hash: Optional[str] = os.environ.get("BUILD_SOURCEVERSION")
+    performance_repo_ci: bool = False
+    use_local_commit_time: bool = False
+    javascript_engine_path: Optional[str] = None
+    maui_version: Optional[str] = None
+    download_pdn: bool = False
+    os_version: Optional[str] = None
+    dotnet_version_link: Optional[str] = None
+    target_csproj: Optional[str] = None
+
+def get_pre_commands(args: RunPerformanceJobArgs, v8_version: str):
+    helix_pre_commands: list[str] = []
+
+    # Increase file handle limit for Alpine: https://github.com/dotnet/runtime/pull/94439
+    if args.os_sub_group == "_musl":
+        helix_pre_commands += ["ulimit -n 4096"]
+
+    # Remember the previous PYTHONPATH that was set so it can be restored in the post commands
+    if args.os_group == "windows":
+        helix_pre_commands += ["set ORIGPYPATH=%PYTHONPATH%"]
+    else:
+        helix_pre_commands += ["export ORIGPYPATH=$PYTHONPATH"]
 
-def run_performance_job(args: RunPerformanceJobArgs):
-    if args.project_file is None:
-        args.project_file = os.path.join(args.performance_repo_dir, "eng", "performance", "helix.proj")
+    # Allow using OpenSSL 1.0.2: https://github.com/dotnet/runtime/pull/60728 
+    # TODO: Is this still needed? 
+    if args.os_group != "windows":
+        helix_pre_commands += ["export CRYPTOGRAPHY_ALLOW_OPENSSL_102=true"]
```

Comment:
At least in current performance micro testing (akadeindexset) on ubuntu, this is not being exported based on the run logs.

Reply:
I think I will leave it in for now, but in a future refactor of this file we can remove it.